### PR TITLE
Download and open attachment

### PR DESCRIPTION
**Pull request by hoangdat**
- [x] iOS
https://user-images.githubusercontent.com/6462404/172882591-5c6d96bd-0c50-46ca-9c91-87d4974b63fb.mp4

- [x] Android

https://user-images.githubusercontent.com/6462404/172882731-77ca3232-e065-44cf-afe7-63ac298bec4c.mp4

### CODE REVIEW

File: core/lib/data/network/download/download_manager.dart [hunk 1]
```diff
@@ -100,4 +105,14 @@ class DownloadManager {
 
     return false;
   }
+
+  MediaType? _extractMediaTypeFromResponse(ResponseBody responseBody) {
+    try {
+      final contentType = responseBody.headers['content-type'];
```
Comment (ManhNTX) [inline core/lib/data/network/download/download_manager.dart:111]: should use Headers.contentTypeHeader replace 'content-type'

File: lib/features/email/data/network/email_api.dart [hunk 2]
```diff
@@ -23,9 +23,9 @@ import 'package:jmap_dart_client/jmap/mail/email/submission/email_submission.dar
 import 'package:jmap_dart_client/jmap/mail/email/submission/email_submission_id.dart';
 import 'package:jmap_dart_client/jmap/mail/email/submission/envelope.dart';
 import 'package:jmap_dart_client/jmap/mail/email/submission/set/set_email_submission_method.dart';
+import 'package:model/model.dart';
```
Comment (dab246) [inline lib/features/email/data/network/email_api.dart:26]: You should import specifically the class to be used. Avoid importing the whole module.

File: core/lib/data/network/download/download_manager.dart [hunk 3]
```diff
@@ -100,4 +105,14 @@ class DownloadManager {
...
+      return MediaType.parse(contentType!.first);
+    } catch (e) {
+      logError('DownloadManager::_extractMediaTypeFromResponse(): $e');
+      return null;
+    }
+  }
 }
```
Comment (ManhNTX) [inline core/lib/data/network/download/download_manager.dart:118]: add line empty at last file

File: core/lib/data/network/download/downloaded_response.dart [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+class DownloadedResponse with EquatableMixin {
+  final String filePath;
+  final MediaType? mediaType;
+
+  DownloadedResponse(this.filePath, {this.mediaType});
+
+  @override
+  List<Object?> get props => [filePath, mediaType];
+}
```
Comment (ManhNTX) [inline core/lib/data/network/download/downloaded_response.dart:12]: add line empty at last file

File: lib/l10n/intl_messages.arb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (ManhNTX) [inline lib/l10n/intl_messages.arb:1]: Re generator this file

File: lib/main/localizations/app_localizations.dart [hunk 6]
```diff
@@ -1441,4 +1441,11 @@ class AppLocalizations {
...
         name: 'ssoNotAvailable');
   }
+
+  String get noPreviewAvailable {
+    return Intl.message(
+      'No preview available',
+      name: 'noPreviewAvailable',
+    );
+  }
 }
```
Comment (ManhNTX) [inline lib/main/localizations/app_localizations.dart:1451]: add line empty at last file

**Review by ManhNTX — CHANGES_REQUESTED** [review]

File: core/lib/data/network/download/download_manager.dart (hunk 3 above)
Comment (dab246) [reply core/lib/data/network/download/download_manager.dart:118]: IMO, it's not necessary to do so.

**Comment by hoangdat** [comment]
Hi @chibenwa, 
On friday: when we connect to
``` 
https://jmap.upn.integration-open-paas.org/.well-known/webfinger?resource=https://jmap.upn.integration-open-paas.org&rel=http://openid.net/specs/connect/1.0/issuer
```
404. -> so we can redirect the app to use `basicAuth` flow.
But today, when connect to it, response:
```
{
  "subject": "https://jmap.upn.integration-open-paas.org",
  "links": [
    {
      "rel": "http://openid.net/specs/connect/1.0/issuer",
      "href": "https://auth-dev.upn.integration-open-paas.org/auth/realms/oidc"
    }
  ]
}
```
So no way to test with `basicAuth`. Can you have any suggestion?

**Review by ManhNTX — APPROVED** [review]
LGTM

**Review by dab246 — APPROVED** [review]

**Pull request merged**